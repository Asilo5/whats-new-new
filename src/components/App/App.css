* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.app {
    display: flex;
    flex-direction: column;
}

h1 {
  text-align: center;
  font-family: 'UnifrakturMaguntia', cursive;
  font-size: 5em;
  margin: 20px;
}


