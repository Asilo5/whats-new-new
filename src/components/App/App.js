import React, { Component } from 'react';
import './App.css';
import NewsContainer from '../NewsContainer/NewsContainer';
import Menu from '../Menu/Menu';
import SearchForm from '../SearchForm/SearchForm';

class App extends Component {
  constructor() {
    super();
    this.state = {
      allNews: {},
      currentNews: [],
      hasError: ''
    }
  }

  componentDidMount() {
    fetch('https://whats-new-api.herokuapp.com/api/v1/news')
    .then(resp => resp.json())
    .then(data => this.setState({ allNews : data, currentNews : data.local }))
    .catch(error => this.setState({ hasError : error }))
  }

  filterNews = (chosenNews) => {
    this.setState({ currentNews: chosenNews })
  }

  searchNews = (searchedNews) => {
    let foundNews = this.state.currentNews.filter((news) => {
      let lowerCaseNews = news.headline.toLowerCase();
      return lowerCaseNews.includes(searchedNews.toLowerCase())
    });
    this.setState({ currentNews: foundNews })
  }

  render () {
    return (
      <section className="app">
        <h1>What's New Times?</h1>
        <Menu allNews={this.state.allNews} filterNews={this.filterNews} />
        <SearchForm searchNews={this.searchNews} />
        <NewsContainer currentNews={this.state.currentNews} />
      </section>
    );
  }
}

export default App;
