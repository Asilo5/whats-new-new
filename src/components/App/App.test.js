import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('should match snapshot with all information passing in correctly', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  })
})
