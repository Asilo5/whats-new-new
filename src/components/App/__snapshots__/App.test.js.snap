// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot with all information passing in correctly 1`] = `
<section
  className="app"
>
  <h1>
    What's New Times?
  </h1>
  <Menu
    allNews={Object {}}
    filterNews={[Function]}
  />
  <SearchForm
    searchNews={[Function]}
  />
  <NewsContainer
    currentNews={Array []}
  />
</section>
`;
